src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String DELIMITER = "///";
+    private static final int INDEX_0 = 0;
+    private static final int INDEX_1 = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] text = readFile(fromFileName);
+        int[] data = calculateDataForReport(text);
+        String dataForPrint = createReport(data);
+        putDataToFile(dataForPrint,toFileName);
+    }
+
+    private String[] readFile(String fileName) {
+        StringBuilder string = new StringBuilder();
+        try {
+            File file = new File(fileName);
+            FileReader fileReader = new FileReader(file);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String text = bufferedReader.readLine();
+            while (text != null) {
+                string.append(text).append(DELIMITER);
+                text = bufferedReader.readLine();
+            }
+            bufferedReader.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t open file!" + fileName, e);
+        }
+        if (string.toString().isEmpty()) {
+            String[] empty = {};
+            return empty;
+        }
+        return string.toString().split(DELIMITER);
+    }
+
+    private int[] calculateDataForReport(String[] str) {
```

Comment:
not a mistake, but it is not obvious what our method returns so I would add some [Javadoc](https://www.tutorialspoint.com/java/java_documentation.htm) on what is this array and what is what by the index of that array element